Feat/add mkdocs

Rakesh-Seenu opened the pull request:
## For authors
![demo](https://github.com/user-attachments/assets/02c3d7cc-a1fc-49bd-8e3c-ece2ccdc1964)

## Description

# Updating the 2nd Commit

# The PR request is being divided into two commit :

Commit 1: Addition of mkdocs.yml for Documentation Automation

This PR introduces the MKDocs configuration (mkdocs.yml) along with the setup for the documentation of the AIAgents4Pharma project. The mkdocs.yml file includes the following:

1`. Site Configuration:
- site_name: The documentation is titled AIAgents4Pharma.
- repo_url and repo_name: Links to the GitHub repository for easy access and reference.

2 . Navigation Structure (nav):
The documentation is organized into multiple sections for better navigation:
- Home: Contains the homepage file home.md.
- Talk2Biomodels:
Overview and models (e.g., System Bio Model, Basico Model).
Tools for interacting with models (e.g., Asking Questions, Model Descriptions, Plotting Figures, Simulating Models).
- Talk2Cells: Provides information about the Talk2Cells module.
- Talk2KnowledgeGraphs: Details about the Talk2KnowledgeGraphs module.
Documentation:
- Operational guides (Ops), with CodeOps and DevOps files.
- Contribution guidelines (Contributing), with a CONTRIBUTING.md file for contributors.

3 . Theme Configuration:
- Custom logo: robot.png (can also be a text emoji like 🤖).
- Icons: Material and FontAwesome icons for annotations and the repository link.
- Light/Dark Mode Toggle: Configured to switch between themes, with custom icons for light and dark modes.

4 . Plugins:
- tags: Enables tagging of documentation content.
- mkdocstrings: For automatic documentation generation from docstrings.

Commit 2 :  will come at the end of the week ( Implementation of Jupyter Notebook for User Guidance)

![image](https://github.com/user-attachments/assets/f59134f0-c1f8-412f-84d8-7cc24f089f0e)

Implemented a Tools.ipynb Jupyter notebook file located in docs/notebooks/talk2biomodels/Tools.ipynb. This notebook provides a step-by-step tutorial covering:

- Search Model: Instructions for searching models in the system.
- Model Description: Guidance on exploring and understanding model details.
- Model Simulation: Steps for simulating models effectively.
- Asking Questions: Tutorials for querying and interacting with models.
- This addition aims to guide users through these processes in an intuitive and structured manner.

## issues #28 

## Type of change
Please delete options that are not relevant.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x]  This change is a documentation update

## How Has This Been Tested?
- [x] Have tested the website by running it locally
To view the documentation locally, run the following command after installing MKDocs and its dependencies:

mkdocs serve

This will start a local development server and allow you to preview the documentation in your browser.

## Checklist
- [x] My code follows the style guidelines mentioned in the Code/DevOps guides
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. MkDocs)
- [x] My changes generate 3 new warnings
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# For reviewers

## Checklist pre-approval
- [ ] Is there enough documentation?  
- [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
- [ ] Does the test(s) successfully test edge/corner cases?
- [ ] Does the PR pass the tests? (if the repository has continuous integration)

## Checklist post-approval
- [ ] Does this PR merge ```develop``` into ```main```? If so, please make sure to add a prefix (feat/fix/chore) and/or a suffix BREAKING CHANGE (if it's a major release) to your commit message.
- [ ] Does this PR close an issue? If so, please make sure to descriptively close this issue when the PR is merged.

## Checklist post-merge
- [ ] When you approve of the PR, merge and close it (Read this [article](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) to know about different merge methods on GitHub)
- [ ] Did this PR merge ```develop``` into ```main``` and is it suppose to run an automated release workflow (if applicable)? If so, please make sure to check under the "Actions" tab to see if the workflow has been initiated, and return later to verify that it has completed successfully.

Rakesh-Seenu commented:
@gurdeep330 Could  you please check this PR draft once .

gurdeep330 reviewed (changes requested):
Good job, Rakesh 👍

I have added a few minor comments. Do you mind explaining the description of the PR to explain what this conmit is about? And also briefly describing the next steps?

I think the icon issue is because of the file path.

Thx!

Rakesh-Seenu commented on new.txt:
i will remove it

gurdeep330 commented:
@dmccloskey do not merge the branch after your review. We will merge after the part 2 of the commit is completed.

dmccloskey commented on mkdocs.yml:
```diff
@@ -0,0 +1,102 @@
+
+theme:
+  logo: ./robot.png
+  #logo: 🤖
```
This line can be deleted.

Rakesh-Seenu replied:
Hi Douglas, I have removed all the redundant lines

dmccloskey commented on docs/Documentation_intro.md:
Should this be moved to the Ops folder?

Rakesh-Seenu replied:
Yes, i have moved it

dmccloskey reviewed (changes requested):
Nice work on getting the documentation up and running 💪.

I only had minor comments that I think can be addressed rather quickly.

Rakesh-Seenu commented:
@dmccloskey Comment 2 : Integration of Tutorial for Models (Issue #28)

This commit completes the second part of Issue #28, which involves integrating the Jupyter Notebook tutorial for the models. Below are the key changes:

Files Changed:

- 1. tutorial.ipynb: Added a Jupyter Notebook to provide a comprehensive tutorial for using the models.
- 2. .yml: Updated configuration to ensure compatibility with the tutorial setup.

I have added Snapshot for the reference

dmccloskey reviewed (approved):
The tutorial notebook looks really nice 🤩.

For future work, I would recommend removing the dependency on Streamlit. @gurdeep330 I believe the migration to LangGraph will solve this?
